packages/tx/src/transaction.ts

```diff
@@ -21,290 +21,255 @@ const N_DIV_2 = new BN('7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46
  * An Ethereum transaction.
  */
 export default class Transaction {
-  public raw!: Buffer[]
-  public nonce!: Buffer
-  public gasLimit!: Buffer
-  public gasPrice!: Buffer
-  public to!: Buffer
-  public value!: Buffer
-  public data!: Buffer
-  public v!: Buffer
-  public r!: Buffer
-  public s!: Buffer
-
-  private _common: Common
-  private _senderPubKey?: Buffer
-  protected _from?: Buffer
+  public readonly common: Common
+  public readonly nonce: BN
+  public readonly gasLimit: BN
+  public readonly gasPrice: BN
+  public readonly to?: Address
+  public readonly value: BN
+  public readonly data: Buffer
+  public readonly v?: BN
+  public readonly r?: BN
+  public readonly s?: BN
+
+  public static fromTxData(txData: TxData, common?: Common) {
```

Comment:
Just realizing while working on a similar transition on the block library: do we want to put `common` here in an options dict? Think this would be more future-proof, will also introduce the like on block since there are options which needed to be passed around.

Reply:
hmm I liked the simplicity of removing the need for TxOptions, but i see your point for future-proofing.

i assume we would do it for the constructor and all the factory methods?

Reply:
Yes, would say so, please also put TxOptions being the last parameter for the constructor for consistency (unless you have got an explicit reason not to).

Yeah, sorry for un-simplify here a bit, but I think this is quite a realistic scenario that we want to add here at some point and it would be a bit a pity if this would be blocked by not have this prepared.